### DISCUSSION TITLE
How to add "Umlaut" characters, i.e., 'ö', 'ä', and 'ü'

### DISCUSSION THREAD

**Discussion by w011e** [nickcoutsos/keymap-editor#146] [Q&A]
Hey there 

Do you know how to add add "Umlaut" characters, i.e., 'ö', 'ä', and 'ü' as well as 'Ö', 'Ä', and 'Ü' respectively? I'd like to add these to the keymap in order to type in German. 

Many thanks 
W011E

**Comment by nickcoutsos** [comment]
Hey, did you mean to close this?

I'd love to get more perspectives from people using my keymap editor and who need support for character sets that I'm not using.

Have a look at [zmk-locale-generator](https://github.com/joelspadin/zmk-locale-generator) -- if you're including headers from that in your build (like `keys_de.h`) then those symbols will become available to you in the app. So you can bind the keycode `DE_U_UMLAUT` and get the symbol `ü`.

I speak basically no German (entschuldigung!) so I don't know if that keycode plus shift necessarily gives you the capitalized version, though I assume it would.

The other thing which is a problem for some people, is that using these keycodes still requires you to set your input language appropriately in the OS. This means the keymap isn't just "automatically portable" between computers and that you can't get characters for multiple locales for free. If that's a problem you may need to rely on whichever method your OS supports for generating these characters (e.g., on macos typing <kbd>alt</kbd>+<kbd>u</kbd> inserts an umlaut, and applies it to the next vowel typed.

Please do let me know if that's of any use to you, or if you need more help setting something up.
